Response Code Field Missing from Resource netscaler_lbmonitor

The definition for netscaler_lbmonitor resource is missing the respcode field.

From the Nitro API docs:
respcode | <String[]> | Read-write | Response  codes for which to mark the service as UP. For any other response code,  the action performed depends on the monitor type. HTTP monitors and  RADIUS monitors mark the service as DOWN, while HTTP-INLINE monitors  perform the action indicated by the Action parameter.

Terraform resource example:

resource "netscaler_lbmonitor" "https_ignore_errors" {
  monitorname = "https_ignore_errors"
  type = "HTTP"
  secure = "YES"
  interval = 5
  resptimeout = 3
  respcode = ["200","301-302","400-404","500-504"]
}

Output of terraform plan:

Error: netscaler_lbmonitor.https_ignore_errors: : invalid or unknown key: respcode

Hi,
I am also looking to use the respcode property
I tried to adjust the code to see if i could incorporate it but when a try to run the build i get this below, any pointers you can provide would be very helpful. I am not proficient in development but am able to understand the code to a degree and make small adjustments.

PS C:\NetscalerProviderCompile> go run main.go
# command-line-arguments
.\main.go:25:3: cannot use "github.com/citrix/terraform-provider-netscaler/netscaler".Provider (type func() "github.com/citrix/terraform-provider-netscaler/vendor/github.com/hashicorp/terraform/terraform".ResourceProvider) as type "github.com/hashicorp/terraform/plugin".ProviderFunc in field value

--- a/netscaler/resource_lbmonitor.go
+++ b/netscaler/resource_lbmonitor.go
@@ -479,6 +479,12 @@ func resourceNetScalerLbmonitor() *schema.Resource {
 				Optional: true,
 				Computed: true,
 			},
+			"respcode": &schema.Schema{
+				Type:     schema.TypeList,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Optional: true,
+				Computed: true,
+			},
 		},
 	}
 }
@@ -590,6 +596,7 @@ func createLbmonitorFunc(d *schema.ResourceData, meta interface{}) error {
 		Validatecred:           d.Get("validatecred").(string),
 		Vendorid:               d.Get("vendorid").(int),
 		Vendorspecificvendorid: d.Get("vendorspecificvendorid").(int),
+		Respcode:               d.Get("respcode").([]interface{}),
 	}
 
 	_, err := client.AddResource(netscaler.Lbmonitor.Type(), lbmonitorName, &lbmonitor)
@@ -711,6 +718,7 @@ func readLbmonitorFunc(d *schema.ResourceData, meta interface{}) error {
 	d.Set("validatecred", data["validatecred"])
 	d.Set("vendorid", data["vendorid"])
 	d.Set("vendorspecificvendorid", data["vendorspecificvendorid"])
+	d.Set("respcode", data["respcode"])
 
 	return nil
 
@@ -1189,6 +1197,17 @@ func updateLbmonitorFunc(d *schema.ResourceData, meta interface{}) error {
 		log.Printf("[DEBUG] netscaler-provider:  Vendorspecificvendorid has changed for lbmonitor %s, starting update", lbmonitorName)
 		lbmonitor.Vendorspecificvendorid = d.Get("vendorspecificvendorid").(int)
 		hasChange = true
+	}
+
+	if d.HasChange("respcode") {
+		log.Printf("[DEBUG] netscaler-provider:  Respcode has changed for lbmonitor %s, starting update", lbmonitorName)
+		_, ok := d.GetOk("respcode")
+		respcode_val := d.Get("respcode").([]interface{})
+
+		if ok {
+			lbmonitor.Respcode = respcode_val
+			hasChange = true
+		}
 	}
 
 	if hasChange {

